robottelo/ssh.py

```diff
@@ -73,18 +76,28 @@ def _get_connection(
         hostname = settings.server.hostname
     if username is None:
         username = settings.server.ssh_username
-    if key_filename is None:
+    if key_filename is None and password is None:
         key_filename = settings.server.ssh_key
+        # If we don't have key even in settings, read password instead:
+        if key_filename is None:
+            password = settings.server.ssh_password
 
     client = _call_paramiko_sshclient()
     client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-    client.connect(
-        hostname=hostname,
-        username=username,
-        key_filename=key_filename,
-        timeout=timeout
-    )
-
+    if key_filename is not None:
+        client.connect(
+            hostname=hostname,
+            username=username,
+            key_filename=key_filename,
+            timeout=timeout
+        )
+    else:
+        client.connect(
+            hostname=hostname,
+            username=username,
+            password=password,
+            timeout=timeout
+        )
```

Comment:
I think this part can be simplified to a single call

Looking paramiko source code I figured out that all params [is passed anyway ](https://github.com/paramiko/paramiko/blob/master/paramiko/client.py#L366 ) following a precedence order.

If key_filename is None (matched the **if** here) it will try to [use the password ](https://github.com/paramiko/paramiko/blob/master/paramiko/client.py#L226-L232)

What about this:

* If key_filename is passed it will try use key_filename (even if password is passed, it is actually the paramiko defaults)
* If password is passed it will be used only if no p_key and no key_filename
* if not password and not key_filename it will try to use the keys from "~./ssh"
* otherwise auth fails

```python
if key_filename is None:
    key_filename = settings.server.ssh_key
if password is None:
    password = settings.server.ssh_password

client = _call_paramiko_sshclient()
client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
client.connect(
    hostname=hostname,
    username=username,
    key_filename=key_filename,
    password=password,
    timeout=timeout
)
```